constants/error.go

```diff
@@ -0,0 +1,48 @@
+package constants
+
+import "errors"
+
+// Custom error types used throughout the application
+var (
+	// ErrUserExistenceFailed indicates a failure in checking user existence.
```

Comment:
No need to add comments for every line, if the understanding of code line is obvious.